pkg/asset/k8s.go

```diff
@@ -116,16 +118,38 @@ const validBootstrapTokenChars = "0123456789abcdefghijklmnopqrstuvwxyz"
 // newBootstrapToken constructs a bootstrap token in conformance with the following format:
 // https://kubernetes.io/docs/admin/bootstrap-tokens/#token-format
 func newBootstrapToken() (id string, secret string, err error) {
-	// Read 6 random bytes for the id and 16 random bytes for the token (see spec for details).
-	token := make([]byte, 6+16)
-	if _, err := rand.Read(token); err != nil {
+	idBytes := make([]byte, 6)
+	secretBytes := make([]byte, 16)
+	tokenLength := len(idBytes) + len(secretBytes)
+	charsetLength := len(validBootstrapTokenChars)
+	size := int(math.Ceil(float64(tokenLength) * math.Log2(float64(charsetLength)) / math.Log2(256)))
+
+	random := make([]byte, size)
+	if _, err := rand.Read(random); err != nil {
 		return "", "", err
 	}
 
-	for i, b := range token {
-		token[i] = validBootstrapTokenChars[int(b)%len(validBootstrapTokenChars)]
+	encodeBytes(validBootstrapTokenChars, random, idBytes, secretBytes)
+	return string(idBytes), string(secretBytes), nil
+}
+
+// encodeBytes encodes the input byte slice into encoded byte slices.
+// The input is treated as a big-endian unsigned integer, and the
+// encoded slices represent the input value in little-endian unsigned
+// base len(charset).  The size of the encoded slices should be set
+// before calling this function; after consuming input, any "extra"
+// encoded slice length will be filled with charset[0].
+func encodeBytes(charset string, input []byte, encoded ...[]byte) {
+	charsetLength := len(charset)
+	inputInt := big.NewInt(0).SetBytes(input)
+	base := big.NewInt(int64(charsetLength))
+	remainder := big.NewInt(0)
+	for _, enc := range encoded {
```

Comment:
Instead of this loop, just encode to a single byte slice and split after like the old code did?